Sources/Commands/SwiftRunCommand.swift

```diff
@@ -272,10 +272,9 @@ public struct SwiftRunCommand: AsyncSwiftCommand {
         }
 
         // If the executable is implicit, search through root products.
-        let rootExecutables = graph.rootPackages
-            .flatMap { $0.products }
-            .filter { $0.type == .executable || $0.type == .snippet }
-            .map { $0.name }
+        let products = graph.rootPackages.flatMap(\.products)
+        let executablesAndSnippets = products.filter { $0.type == .executable || $0.type == .snippet }
```

Comment:
I think this is bit (`==` and `||` operators) are most likely the worst offender here, you can keep the chaining but replace use of `==`.

Reply:
Looks like simply explicitly typing `.executable` and `.snippet` fixed the slowdown